bee-api/bee-rest-api/src/types/dtos.rs

```diff
@@ -536,6 +675,497 @@ pub struct ReferenceUnlockBlockDto {
+        Self(value.to_string())
+    }
+}
+
+impl TryFrom<&AliasIdDto> for AliasId {
+    type Error = Error;
+
+    fn try_from(value: &AliasIdDto) -> Result<Self, Self::Error> {
+        value
+            .0
+            .parse::<AliasId>()
+            .map_err(|_| Error::InvalidSemanticField("alias id"))
+    }
+}
+
+/// Describes a foundry output that is controlled by an alias.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct FoundryOutputDto {
+    #[serde(rename = "type")]
+    pub kind: u8,
+    // Deposit address of the output.
+    address: AddressDto,
+    // Amount of IOTA tokens held by the output.
+    amount: u64,
+    // Native tokens held by the output.
+    #[serde(rename = "nativeTokens")]
+    native_tokens: Vec<NativeTokenDto>,
+    // The serial number of the foundry with respect to the controlling alias.
+    #[serde(rename = "serialNumber")]
+    serial_number: u32,
+    // Data that is always the last 12 bytes of ID of the tokens produced by this foundry.
+    #[serde(rename = "tokenTag")]
+    token_tag: String,
+    // Circulating supply of tokens controlled by this foundry.
+    #[serde(rename = "circulatingSupply")]
+    circulating_supply: U256Dto,
+    // Maximum supply of tokens controlled by this foundry.
+    #[serde(rename = "maximumSupply")]
+    maximum_supply: U256Dto,
+    #[serde(rename = "tokenScheme")]
+    token_scheme: TokenSchemeDto,
+    #[serde(rename = "blocks")]
+    feature_blocks: Vec<FeatureBlockDto>,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub enum TokenSchemeDto {
+    Simple = 0,
+}
+
+impl From<&FoundryOutput> for FoundryOutputDto {
+    fn from(value: &FoundryOutput) -> Self {
+        Self {
+            kind: FoundryOutput::KIND,
+            address: value.address().into(),
+            amount: value.amount(),
+            native_tokens: value.native_tokens().iter().map(Into::into).collect::<_>(),
+            serial_number: value.serial_number(),
+            token_tag: hex::encode(value.token_tag()),
+            circulating_supply: U256Dto(value.circulating_supply().to_string()),
+            maximum_supply: U256Dto(value.maximum_supply().to_string()),
+            token_scheme: match value.token_scheme() {
+                TokenScheme::Simple => TokenSchemeDto::Simple,
+            },
+            feature_blocks: value.feature_blocks().iter().map(Into::into).collect::<_>(),
+        }
+    }
+}
+
+impl TryFrom<&FoundryOutputDto> for FoundryOutput {
+    type Error = Error;
+
+    fn try_from(value: &FoundryOutputDto) -> Result<Self, Self::Error> {
+        let mut builder = FoundryOutputBuilder::new(
+            (&value.address).try_into()?,
+            value.amount,
+            value.serial_number,
+            {
+                let mut decoded_token_tag = [0u8; 12];
+                hex::decode_to_slice(&value.token_tag, &mut decoded_token_tag as &mut [u8])
+                    .map_err(|_| Error::InvalidSyntaxField("token_tag"))?;
```

Comment:
The `Syntax` in `InvalidSyntaxField` refers to a bad parse? I find the name of the error somewhat confusing. Why not just `InvalidField`?

Reply:
`InvalidSyntaxField` assumes the existence of `InvalidSemanticField`, if we don't have it, I'd just drop the `Syntax` part.

Reply:
Since we have some existing variant `InvalidSemanticField` having some `InvalidSyntaxField` makes sense too